pkg/scheduler/objects/queue_events.go

```diff
@@ -0,0 +1,127 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package objects
+
+import (
+	"github.com/apache/yunikorn-core/pkg/common"
+	"github.com/apache/yunikorn-core/pkg/events"
+	"github.com/apache/yunikorn-scheduler-interface/lib/go/si"
+)
+
+type queueEvents struct {
+	enabled     bool
+	eventSystem events.EventSystem
+	queue       *Queue
+}
+
+func (q *queueEvents) sendNewQueueEvent() {
+	if !q.enabled {
+		return
+	}
+
+	var detail si.EventRecord_ChangeDetail
+	if q.queue.IsManaged() {
+		detail = si.EventRecord_DETAILS_NONE
+	} else {
+		detail = si.EventRecord_QUEUE_DYNAMIC
+	}
```

Comment:
detail could be assigned with most frequently used value by default and change it only when necessary.

Reply:
I think you missed this place.